truss/truss_config.py

```diff
@@ -171,6 +172,98 @@ class ModelServer(Enum):
     TRT_LLM = "TRT_LLM"
 
 
+class TRTLLMModelArchitecture(Enum):
+    LLAMA: str = "llama"
+    MISTRAL: str = "mistral"
+
+
+class TRTLLMQuantizationType(Enum):
+    NO_QUANT: str = "no_quant"
+    WEIGHTS_ONLY: str = "weights_only"
+    WEIGHTS_KV_INT8: str = "weights_kv_int8"
+    SMOOTH_QUANT: str = "smooth_quant"
+
+
+class TRTLLMBuildConfiguration(BaseModel):
+    huggingface_ckpt_repository: str
+    base_model_architecture: TRTLLMModelArchitecture
+    max_input_len: int
+    max_output_len: int
+    max_batch_size: int
+    quantization_type: TRTLLMQuantizationType = TRTLLMQuantizationType.NO_QUANT
+    gather_all_token_logits: bool = False
+    multi_block_mode: bool = False
+    calibrate_kv_cache: bool = False
+
+    def from_dict(d):
```

Comment:
do you have to write this? Doesn't pydantic take care of it for you when you ` TRTLLMBuildConfiguration(**d)`

Reply:
We needed some method for the `transform_optional` invocation [here](https://github.com/basetenlabs/truss/pull/825/files#diff-21d89478b5239675a83eba81b5c7c004d1aa27002e11c93b64fefd164d0dd193R612). I found a way around it with lambdas which is cleaner.